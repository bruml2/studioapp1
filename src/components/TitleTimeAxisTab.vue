<template>
  <!-- TitleTimeAxisTab.vue  v0.2  Apr 8 -->
  <v-form>
    <v-container>
      <v-card width="50rem" class="my-3 mx-auto" color="blue-grey lighten-5" raised>
        <v-card-title>
          <h1 class="headline font-weight-bold">Title/subtitle and Time Axis</h1>
        </v-card-title>
        <v-card-text>
          <!-- <p class="title">The timeline's main title: for example, "United States History: "</p> -->
          <v-text-field label="Title" v-model="title" outlined dense />
          <!-- <p class="title">The timeline's subtitle: for example, "the 20th Century"</p> -->
          <v-text-field label="Subtitle" v-model="subtitle" placeholder="type here" outlined dense />

          <div id="tvHeader" class="tvHeader">
            <span class="tlTitle">{{ title }}</span> &nbsp; &nbsp;
              <span class="tlSubtitle">{{ subtitle }}</span>
          </div>

          <v-row>
            <v-col cols="2">
              <v-text-field label="Start Year" v-model.number="startYear" outlined dense width="10rem" />
            </v-col>
            <v-col cols="2">
              <v-text-field label="Stop Year" v-model.number="stopYear" outlined dense />
            </v-col>
            <v-col cols="8">
              Tick Interval:
              <v-radio-group v-model="tickInterval" row>
                <v-radio
                  v-for="n in [5, 10, 20, 25, 50, 100]"
                  :key="n"
                  :label="n.toString()"
                  :value="n"
                ></v-radio>
              </v-radio-group>
            </v-col>
          </v-row>
          <TimeAxisView :timeAxisPropObj="timeAxisPropObj" />
        </v-card-text>
      </v-card>
    </v-container>
  </v-form>
</template>

<script>
import TimeAxisView from "../../../../HistoryTimelineStudio/timelineviewcomp/src/components/TimeAxisView.vue"

export default {
  name: 'TitleTimeAxisTab',
  components: { TimeAxisView },
  props: ["tl"], // empty object;
	data() {
		return {
      title: "",
      subtitle: "",
      startYear: undefined,
      stopYear: undefined,
      tickInterval: undefined
		}
  },
  computed: {
    timeAxisPropObj() {
      // let svgWidthAdj = document.getElementsByClassName('tvHeader')[0].offsetWidth
      // console.log(svgWidthAdj)
      return {
        startYear: this.startYear,
        stopYear:  this.stopYear,
        tickInterval: this.tickInterval,
        svgWidth: 700
      }
    }
  }
}
</script>

<style>
.tvHeader {
	font-family: Palatino, "New Time Roman", serif;
  position: relative; /* parent container for positioning */
  box-sizing: border-box;
  width: 98%;
  margin: 10px auto;
  padding: 17px 20px 10px; /* centered visually */
  text-align: left;
  font-weight: bold;
  border: 3px solid #C11B17;
}
.tlTitle {
  font-size: 24px;
  color: #0404B4;
}
.tlSubtitle {
  font-size: 20px;
}
</style>
