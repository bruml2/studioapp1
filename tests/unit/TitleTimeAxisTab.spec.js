// April 9
// this test file does NOT use localVue;
// as stated in https://vuetifyjs.com/en/getting-started/unit-testing/;
import Vue from 'vue'
import Vuetify from 'vuetify'
Vue.use(Vuetify)
// shallowMount stubs out any child components; here TimeAxisView (tested separately);
import { shallowMount } from '@vue/test-utils'
import TitleTimeAxisTab from "@/components/TitleTimeAxisTab.vue"

const shallowMountFn = function () {
  // let vuetify = new Vuetify()
  const wrapper = shallowMount(TitleTimeAxisTab, {
    // vuetify,
    propsData: {
      tl: {}
      // tl: { title: "Test Title", startYear: 1900, stopYear: 2000, tickInterval: 20 }
    }
  })
  return wrapper;
}

describe('TitleTimeAxisTab.vue', () => {
  /*
  it('should match previous snapshot', () => {
    const wrapper = shallowMountFn()
    expect(wrapper.html()).toMatchSnapshot()
  })
  */

  it('should have h1 text', () => {
    const wrapper = shallowMountFn({})
    expect(wrapper.find('h1').text()).toBe('Title/subtitle and Time Axis')
  })

  it('should have empty text input for Title', () => {
    const wrapper = shallowMountFn({})
    const titleinput = wrapper.find('[data-j="title"]')
    // begins empty;
    expect(titleinput.text()).toMatch('')
  })

  it('input value should appear in v-model data', () => {
    const wrapper = shallowMountFn({})
    // const titleinput = wrapper.find('[data-testtitle]')
    // titleinput.setValue('Title For Test')
    wrapper.find('[data-j="title"]').setValue('Title For Test')
    expect(wrapper.vm.title).toBe('Title For Test')
  })

})
  /*
  it('should emit an event when the action v-btn is clicked', () => {
    const wrapper = shallowMountFn({propsData: { title: 'Foobar' } })
    const event = jest.fn()
    const button = wrapper.find('.v-btn')
    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    wrapper.vm.$on('action-btn:clicked', event)

    expect(event).toHaveBeenCalledTimes(0)

    // Simulate a click on the button
    button.trigger('click')

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1)
  })
})
*/
/*
describe('TitleTimeAxisTab.vue', () => {
  let vuetify

  // more stuff

  it('renders props.msg when passed', () => {
    const tl = { title: "Test Title", startYear: 1900, stopYear: 2000 }
    const wrapper = shallowMount(TitleTimeAxisTab, {
      propsData: { tl }
    })
    expect(wrapper.text()).toMatch(msg)
  })
})
*/
/* import TimeAxisView from "../../../studioTimelineView/src/components/TimeAxisView.vue"
*/
