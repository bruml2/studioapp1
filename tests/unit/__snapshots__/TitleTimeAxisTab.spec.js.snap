// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleTimeAxisTab.vue should have a custom title and match snapshot 1`] = `
<v-form>
  <v-container>
    <v-card width="50rem" color="blue-grey lighten-5" raised="" class="my-3 mx-auto">
      <v-card-title>
        <h1 class="headline font-weight-bold">Titles and Time Axis</h1>
      </v-card-title>
      <v-card-text>
        <v-text-field label="Title" outlined="" dense=""></v-text-field>
        <v-text-field label="Subtitle" placeholder="type here" outlined="" dense=""></v-text-field>
        <div id="tvHeader" class="tvHeader"><span class="tlTitle">Test Title</span> &nbsp; &nbsp;
          <span class="tlSubtitle"></span></div>
        <v-row>
          <v-col cols="2">
            <v-text-field label="Start Year" outlined="" dense="" width="10rem"></v-text-field>
          </v-col>
          <v-col cols="2">
            <v-text-field label="Stop Year" outlined="" dense=""></v-text-field>
          </v-col>
          <v-col cols="8">
            Tick Interval:
            <v-radio-group row="">
              <v-radio label="5" value="5"></v-radio>
              <v-radio label="10" value="10"></v-radio>
              <v-radio label="20" value="20"></v-radio>
              <v-radio label="25" value="25"></v-radio>
              <v-radio label="50" value="50"></v-radio>
              <v-radio label="100" value="100"></v-radio>
            </v-radio-group>
          </v-col>
        </v-row>
        <timeaxisview-stub timeaxispropobj="[object Object]"></timeaxisview-stub>
      </v-card-text>
    </v-card>
  </v-container>
</v-form>
`;
